account_manager/src/validator/exit.rs

```diff
@@ -0,0 +1,440 @@
+            Ok(index) => {
+                if let Err(e) = publish_voluntary_exit::<E>(
+                    &keypair,
+                    client,
+                    spec,
+                    stdin_inputs,
+                    epoch,
+                    index,
+                    &fork,
+                    genesis_data.genesis_validators_root,
+                )
+                .await
+                {
+                    eprintln!(
+                        "Failed to publish voluntary exit for validator {:?}: {}",
+                        validator_dir.dir(),
+                        e
+                    );
+                }
+            }
+            Err(e) => eprintln!("{}", e),
+        }
+    }
+    Ok(())
+}
+
+/// Constructs a `VoluntaryExit` object for a given validator and signs it using the given bls keypair.
+/// Publishes the voluntary exit to the beacon chain using the beacon node endpoint.
+#[allow(clippy::too_many_arguments)]
+async fn publish_voluntary_exit<E: EthSpec>(
+    keypair: &Keypair,
+    client: &BeaconNodeHttpClient,
+    spec: &ChainSpec,
+    stdin_inputs: bool,
+    epoch: Epoch,
+    validator_index: u64,
+    fork: &Fork,
+    genesis_validators_root: Hash256,
+) -> Result<(), String> {
+    let voluntary_exit = VoluntaryExit {
+        epoch,
+        validator_index,
+    };
+
+    let signed_voluntary_exit =
+        voluntary_exit.sign(&keypair.sk, &fork, genesis_validators_root, spec);
+
+    eprintln!(
+        "Publishing a voluntary exit for validator: {} \n",
+        keypair.pk
+    );
+    eprintln!("WARNING: THIS IS AN IRREVERSIBLE OPERATION\n");
+    eprintln!("{}\n", PROMPT);
+    eprintln!(
+        "PLEASE VISIT {} TO MAKE SURE YOU UNDERSTAND THE IMPLICATIONS OF A VOLUNTARY EXIT.",
+        WEBSITE_URL
+    );
+    eprintln!("Enter the exit phrase from the above URL to confirm the voluntary exit: ");
+
+    let confirmation = account_utils::read_input_from_user(stdin_inputs)?;
+    if confirmation == CONFIRMATION_PHRASE {
+        // Verify and publish the voluntary exit to network
+        client
+            .post_beacon_pool_voluntary_exits(&signed_voluntary_exit)
+            .await
+            .map_err(|e| format!("Failed to publish voluntary exit: {}", e))?;
+        tokio::time::delay_for(std::time::Duration::from_secs(1)).await; // Provides nicer UX.
+        eprintln!(
+            "Successfully validated and published voluntary exit for validator {}",
+            keypair.pk
+        );
+    } else {
+        eprintln!(
+            "Did not publish voluntary exit for validator {}. Please check that you entered the correct passphrase.",
+            keypair.pk
+        );
+    }
+    Ok(())
+}
+
+/// Get the validator index af given the validator public key by querying the beacon node endpoint.
+/// Returns an error if the beacon endpoint returns an error or given validator is not eligible for an exit.
+async fn get_validator_index<E: EthSpec>(
```

Comment:
I think we should call this `get_validator_index_for_exit` since we also do verification in here to see if they're eligible.